### PR TITLE
Changes to make the component Polymer 2 compliant.

### PR DESCRIPTION
**Pull request by alexanderson1993**
v1 Story: S-90599
Task: TK-211515

## Changes
* Modified the `<content>` tag to be `<slot>` as explained in the [Polymer Hybrid Element Guide](https://www.polymer-project.org/2.0/docs/devguide/hybrid-elements#working-with-dom)
* Used `this.querySelector()` instead of `this.$.` to access internal elements (where necessary)
## To-Dos
- [ ] Tests
- [ ] Update demo
- [ ] Update documentation & README
- [ ] Increment bower.json version

### CODE REVIEW

**Comment by alexanderson1993** [comment]
I'm not sure what is causing the travis build to break. Is that something I did?

**Comment by skye2k2** [comment]
> Encrypted environment variables have been removed for security reasons.
See https://docs.travis-ci.com/user/pull-requests/#Pull-Requests-and-Security-Restrictions
Pull requests sent from forked repositories do not have access to encrypted variables or data.

Not overly helpful of Travis CI...

**Comment by alexanderson1993** [comment]
That's a bummer. Any chance you could merge my fork into a branch of the repo itself and then issue a pull request of that branch? Or give me write access to this repo?

File: fs-dialog.html [hunk 1]
```diff
@@ -158,13 +158,13 @@
           _overlay: {
             type: Object,
             value: function(){
-              return this.$._overlay;
+              return this.querySelector('#_content');
```
Comment (Norman-Amy) [inline fs-dialog.html]: This looks like the selectors need to be reversed. This one should be for `_overlay` not `_content`.

**Comment by Norman-Amy** [comment]
@alexanderson1993 For now, we'd like to keep using forks for PRs. Although we want fs-dialog to be more shareable, we don't feel it's quite there yet to be open for other teams to start committing PRs without us reviewing them first. We use this component a lot in our apps and have to test any changes thoroughly to ensure things are where they still need to be. Don't get me wrong, we love having PRs, so thank you for submitting this. :)

Also, don't worry about the Travis-ci builds. We expect them to fail when PRs are made across forks, but we will pull down your changes and test them locally to make sure the unit tests still pass.

**Comment by alexanderson1993** [comment]
Awesome @Norman-Amy. Let me know if there are any other changes that need to be made. I intend to put up several pull requests over the next few days to add features I need for an app I'm working on. Nothing too crazy - just the ability to position the component based on a target, optionally making the component go full-width on mobile screen sizes, etc.

**Comment by Norman-Amy** [comment]
Sorry about the delay. There are a few unit tests breaking consistently locally. I'll be able to look into them more on Monday.

**Comment by Norman-Amy** [comment]
@alexanderson1993 We've changed the repo so that you should now have write access to all branches _except_ for master, so if you could move this to another PR so that the Travis-ci build will run, that'd be great. 

Also, please address the breaking unit tests. I'm seeing `this.querySelector(#_overlay);` come back as `undefined` for one of the failing tests. If you need help running them, let me know.

**Comment by alexanderson1993** [comment]
I'm having trouble pushing to a new branch on the main repo. I'm getting 403 errors.

**Comment by skye2k2** [comment]
Should work, now. Sorry.

**Comment by alexanderson1993** [comment]
Thanks. I'll get that other PR up ASAP.

**Pull request closed without merging**